unblob/handlers/archive/netgear/trx.py

```diff
@@ -0,0 +1,115 @@
+# import binascii
+import binascii
+import io
+from pathlib import Path
+from typing import Optional
+
+from dissect.cstruct import Instance
+from structlog import get_logger
+
+from unblob.extractor import carve_chunk_to_file
+from unblob.file_utils import Endian, File, InvalidInputFormat, StructParser
+from unblob.models import Chunk, Extractor, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+CRC_CONTENT_OFFSET = 12
+
+TRX_V1_C_DEFINITION = r"""
+    typedef struct trx_header {
+        uint32 magic;              /* "HDR0" */
+        uint32 len;                /* header size + data */
+        uint32 crc32;              /* 32-bit CRC from flag_version to end of file */
+        uint16 flags;
+        uint16 version;
+        uint32 offsets[3];         /* Offsets of partitions from start of header */
+    } trx_header_t;
+"""
+
+TRX_V2_C_DEFINITION = r"""
+    typedef struct trx_header {
+        uint32 magic;              /* "HDR0" */
+        uint32 len;                /* header size + data */
+        uint32 crc32;              /* 32-bit CRC from flag_version to end of file */
+        uint16 flags;
+        uint16 version;
+        uint32 offsets[4];         /* Offsets of partitions from start of header */
+    } trx_header_t;
+"""
+
+
+class TRXExtractor(Extractor):
+    def __init__(self, c_definitions: bytes):
+        self._struct_parser = StructParser(c_definitions)
+
+    def extract(self, inpath: Path, outdir: Path):
+        with File.from_path(inpath) as file:
+            header = self._struct_parser.parse("trx_header_t", file, Endian.LITTLE)
+            file.seek(0, io.SEEK_END)
+            eof = file.tell()
+            i = 0
+            offsets = sorted(
+                [offset for offset in header.offsets + [eof] if offset > 0]
+            )
+            for start_offset, end_offset in zip(offsets, offsets[1:]):
+                chunk = Chunk(start_offset=start_offset, end_offset=end_offset)
+                carve_chunk_to_file(outdir.joinpath(Path(f"part{i}")), file, chunk)
+                i += 1
+
+
+class NetgearTRXBase(StructHandler):
+    HEADER_STRUCT = "trx_header_t"
+
+    def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]:
+        header = self.parse_header(file, endian=Endian.LITTLE)
+
+        if not self.is_valid_header(header):
+            raise InvalidInputFormat("Invalid TRX header.")
+
+        if header.version == 1 and not self._is_crc_valid(file, start_offset, header):
```

Comment:
Investigate why the CRC32 does not work with version 2.

Reply:
The utility used to create the sample files created wrong headers, the code worked for both versions initially.